src/appleseed/renderer/kernel/rendering/pixelrendererbase.cpp

```diff
@@ -116,6 +117,10 @@ void PixelRendererBase::on_tile_end(
                     tile.get_pixel(x, y, color);
                     color.rgb().set(0.2f * luminance(color.rgb()));     // 20% of luminance
                 }
+                else if (sample_state[0] == NoSampleHint)
```

Comment:
Simplify the logic, maybe with a `switch` where `color` gets assigned once in each case, instead of having a default value that later gets overwritten or altered.

Reply:
Yes